### ISSUE TITLE
Improper frame rendering

### ISSUE THREAD

**Issue by D3adTrace** [Swordfish90/cool-retro-term#703]
I've updated retro term, and saw this
![retro-term-issue](https://user-images.githubusercontent.com/100424861/155740673-9ceec857-e321-4eeb-94ce-68fbef95687a.png)

On previous version (1.1.1-2 in arch linux) everything is okay

Here's --verbose output
```
$ cool-retro-term --verbose
Using a variable-width font in the terminal.  This may cause performance degradation and display/alignment errors.
Both point size and pixel size set. Using pixel size.
Using a variable-width font in the terminal.  This may cause performance degradation and display/alignment errors.
qml: Loading settings: {
  "fps": 0,
  "x": 4,
  "y": 22,
  "width": 1912,
  "height": 1054,
  "windowScaling": 1,
  "showTerminalSize": true,
  "fontScaling": 1,
  "fontNames": [
    "IBM_3278",
    "COMMODORE_PET",
    "COMMODORE_PET"
  ],
  "showMenubar": false,
  "bloomQuality": 1,
  "burnInQuality": 0.5,
  "useCustomCommand": false,
  "customCommand": ""
}{
  "backgroundColor": "#000000",
  "fontColor": "#0ccc68",
  "flickering": 0,
  "horizontalSync": 0,
  "staticNoise": 0,
  "chromaColor": 0,
  "saturationColor": 0,
  "screenCurvature": 0.5198,
  "glowingLine": 0,
  "burnIn": 0.6,
  "bloom": 0.2969,
  "rasterization": 0,
  "jitter": 0,
  "rbgShift": 0,
  "brightness": 0.5,
  "contrast": 0.85,
  "ambientLight": 0.1,
  "windowOpacity": 1,
  "fontName": "IBM_3278",
  "fontWidth": 1,
  "margin": 0.5556,
  "blinkingCursor": false,
  "frameMargin": 0.3014
}
qml: Storing settings: {
  "fps": 0,
  "x": 963,
  "y": 22,
  "width": 953,
  "height": 1054,
  "windowScaling": 1,
  "showTerminalSize": true,
  "fontScaling": 1,
  "fontNames": [
    "IBM_3278",
    "COMMODORE_PET",
    "COMMODORE_PET"
  ],
  "showMenubar": false,
  "bloomQuality": 1,
  "burnInQuality": 0.5,
  "useCustomCommand": false,
  "customCommand": ""
}
qml: Storing profile: {
  "backgroundColor": "#000000",
  "fontColor": "#0ccc68",
  "flickering": 0,
  "horizontalSync": 0,
  "staticNoise": 0,
  "chromaColor": 0,
  "saturationColor": 0,
  "screenCurvature": 0.5198,
  "glowingLine": 0,
  "burnIn": 0.6,
  "bloom": 0.2969,
  "rasterization": 0,
  "jitter": 0,
  "rbgShift": 0,
  "brightness": 0.5,
  "contrast": 0.85,
  "ambientLight": 0.1,
  "windowOpacity": 1,
  "fontName": "IBM_3278",
  "fontWidth": 1,
  "margin": 0.5556,
  "blinkingCursor": false,
  "frameMargin": 0.3014
}
```

**Comment by ia3crt** [comment]
I have the same issue. Digging up found there was the same bug in 2019. It is related to Nvidia. I tested it to use the Igpu on my laptop instead of the nvidia drivers, and sure enough cool-retro-term rendered fine.

**Comment by D3adTrace** [comment]
> I have the same issue. Digging up found there was the same bug in 2019. It is related to Nvidia. I tested it to use the Igpu on my laptop instead of the nvidia drivers, and sure enough cool-retro-term rendered fine.

You mean nouveau drivers?

**Comment by KickingAnimal** [comment]
Also having this issue on Arch with official nvidia drivers.

**Comment by ia3crt** [comment]
> 

No, sorry for not specifying. So far Nouveau seems to render fine, same as amdgpu and intel. In my case, the problem seems exclusive to the privative Nvidia drivers...  I am not 100% sure because amdgpu was tested on a different machine, and nouveau on a different distro (but same 5.16.11 kernel)

**Comment by D3adTrace** [comment]
> > 
> 
> No, sorry for not specifying. So far Nouveau seems to render fine, same as amdgpu and intel. In my case, the problem seems exclusive to the privative Nvidia drivers... I am not 100% sure because amdgpu was tested on a different machine, and nouveau on a different distro (but same 5.16.11 kernel)

I would install nouveau drivers, but it doesn't support power management. Also, if you saw neofetch output, you know that my GPU is discrete, and its name is gt1030

**Comment by njhanley** [comment]
4abbe33 introduced this issue.

**Comment by Adm-Beck** [comment]
Same issue on Ryzen AMD integrated card

**Comment by vananasun** [comment]
bump, same issue while using 1.2.0 on Manjaro with NVIDIA proprietary drivers on an RTX 3060ti

**Comment by 0atman** [comment]
Same problem here on Ubuntu 22.04 with propitiatory nvidia drivers.

Will be trying nouveau as a fix, thanks!

**Comment by 0atman** [comment]
Nouveau did fix the frame, but very often locks up my computer when changing settings XD

**Comment by WhiskeyTuesday** [comment]
If you're okay with recompiling from source there's a workable temporary fix in one of the other issues:
https://github.com/Swordfish90/cool-retro-term/issues/739#issuecomment-1347784249

**Comment by 5150-ibm** [comment]
Still no fix? Was this project abandoned

**Comment by The-Toblin** [comment]
Certainly looks that way. I've since switched GPUs and it doesn't affect AMD cards. 

In addition, while the terminal is cool and gimmicky, it's far too awkward to use in any kind of real scenario, so I just...kinda don't use it anymore. I feel like that's a common theme and maybe the dev feels the same.

**Comment by 0atman** [comment]
I too fixed this with hardware: Last year I traded my nvidia for an equivalent 2nd hand AMD card and have never looked back, Linux is *so* much better now - and I've realised my monitor supports HDR, something that just... didn't work on nvidia.

Some projects (*cough* [sway](https://github.com/swaywm/sway/issues/6797)) straight-up don't accept support requests for nvidia hardware because they're so hostile to the linux desktop community (despite benefiting from the linux *server* for all their genai slop money)

I still use CRT professionally every week to [produce scifi visualisations](https://www.youtube.com/watch?v=p3bDE9kszMc&list=PL95NP4bDITAln7fq-cCqzOFE15UvVthuL&index=2), and I'm very grateful it exists, thanks @Swordfish90 !

**Comment by bmurphy1976** [comment]
If you are looking for an alternative Ghostty has shader support.  The shaders aren't quite as cool as the full experience you can get with CRT, but they go pretty far and it's a much more full featured Terminal Emulator.

**Comment by dhanielrs** [comment]
Still happening with me :/ 
RTX 5060 and Zorin OS 22

**Comment by Digital-Sequence** [comment]
> Still happening with me :/ RTX 5060 and Zorin OS 22

Arch linux, RTX 5060 ti, driver nvidia-open, app version is 2.0.0beta2-1
No problems, everything renders fine. Though, one detail I noticed is, frame is too sharp, comparing to the screenshots on repository

<img width="261" height="238" alt="Image" src="https://github.com/user-attachments/assets/a5dc9ef3-b61e-435c-9fe1-d58e3fa1c5e0" />
